addons/l10n_ae/views/report_invoice_templates.xml

```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="report_invoice_document" inherit_id="account.report_invoice_document">
+        <xpath expr="//p[@name='payment_term']" position="after">
+            <p t-if="o.env.ref('l10n_ae.gcc_countries_group') in o.partner_id.country_id.country_group_ids and o.company_id.country_id.code == 'AE'">
+                Supply between <b>United Arad Emirates</b> and <b><span t-field="o.partner_id.country_id.name"/></b>
+            </p>
+        </xpath>
+        <xpath expr="//h2" position="replace">
+            <h2>
+                <t t-if="o.env.ref('l10n_ae.gcc_countries_group') in o.partner_id.country_id.country_group_ids and o.company_id.country_id.code == 'AE'">
```

Comment:
AFAIK the FTA spec says the term is Tax Invoice / Tax Credit Note for all validated invoices/refunds issued by a UAE company, regardless of the partner country - why limit it to GCC countries?

Also this inheritance mechanism breaks modularity of reports, now other inherits by other l10n modules won't work because you have duplicate the original code. Replacing the whole h2 is unclean. I think we need a cleaner mechanism than that, maybe similar to the VAT label system we have for localizing Tax ID. Please discuss with @qdp-odoo / @jco-odoo

Reply:
given that the issue seems solved, I suppose we don't need such a system... 

Note that I'm saying without having understood the base use case as the previous diff is already not available anymore

Reply:
@qdp-odoo the problem still exists: they are replacing the whole h2 title with another h2 section. If multiple localisations use the same technique only the last one will apply, so it's broken by design.